assume/units/dst_components.py

```diff
@@ -2,782 +2,1592 @@
-        # Convert minimum operating time to the time unit of your choice
-        min_operating_time_units = int(min_operating_time / delta_t)
-
-        # Check for valid indexing
-        if t < min_operating_time_units:
-            if t - min_operating_time_units + 1 < 0:
+        # Define variables
+        model_part.power = pyo.Var(
+            self.time_steps,
+            within=pyo.NonNegativeReals,
+            bounds=(0, self.max_power),
+        )
+
+        # Define constraints
+
+        # Predefined power profile constraint
+        if self.power_profile is not None:
+            if not isinstance(self.power_profile, pd.Series):
+                raise TypeError(
+                    "Residential PV `power_profile` must be a pandas Series."
+                )
+            if not all(t in self.power_profile.index for t in self.time_steps):
                 raise ValueError(
-                    f"Invalid min_operating_time: {min_operating_time} exceeds available time steps. "
-                    "Ensure min_operating_time is compatible with the available time steps."
+                    "All `time_steps` must be present in residential PV `power_profile` index."
                 )
-            return (
-                sum(
-                    b.power_dri[i]
-                    for i in range(t - min_operating_time_units + 1, t + 1)
+
+            @model_part.Constraint(self.time_steps)
+            def power_profile_constraint(b, t):
+                """
+                Ensures the PV follows the predefined power profile.
+                """
+                return b.power[t] == self.power_profile[t]
+
+        # Availability profile constraints
+        if self.availability_profile is not None:
+            if not isinstance(self.availability_profile, pd.Series):
+                raise TypeError(
+                    "Residential PV `availability_profile` must be a pandas Series."
                 )
-                >= min_operating_time_units * b.power_dri[t]
-            )
-        else:
-            return (
-                sum(
-                    b.power_dri[i]
-                    for i in range(t - min_operating_time_units + 1, t + 1)
+            if not all(t in self.availability_profile.index for t in self.time_steps):
+                raise ValueError(
+                    "All `time_steps` must be present in residential PV `availability_profile` index."
                 )
-                >= min_operating_time_units * b.power_dri[t]
-            )
 
-    @model_part.Constraint(time_steps)
-    def min_down_time_dri_constraint(b, t):
-        if t == 0:
-            return pyo.Constraint.Skip
+            @model_part.Constraint(self.time_steps)
+            def availability_pv_constraint(b, t):
+                """
+                Ensures the PV operates only during available periods.
+                """
+                return b.power[t] <= self.availability_profile[t] * b.max_power
```

Comment:
Why not `==` to force power to be the available one? Or do you think about a controlled PV system where the power can be regulated?

Reply:
if we use == we don't allow any flexibility in the system. using this form the household can also deliver flexibility in the future by curtailing PV